### PR TITLE
glfw: ensure git present

### PR DESCRIPTION
**Pull request by d3m1gd**
- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

**Review by iddev5 — COMMENTED** [review]
Top level build.zig also needs this check as it contains ensureSubmodule()

**Comment by d3m1gd** [comment]
it might not be needed there, as glfw will be called anyaway, and missing git will be discovered

**Comment by iddev5** [comment]
That won't work. Non-native (wasm) builds don't call glfw.link().

File: build.zig [hunk 1]
```diff
@@ -211,10 +211,31 @@ fn ensureDependencySubmodule(allocator: std.mem.Allocator, path: []const u8) !vo
     if (std.process.getEnvVarOwned(allocator, "NO_ENSURE_SUBMODULES")) |no_ensure_submodules| {
         if (std.mem.eql(u8, no_ensure_submodules, "true")) return;
     } else |_| {}
+    ensureGit(allocator);
```
Comment (iddev5) [inline build.zig:214]: I think this should be rather put here: https://github.com/hexops/mach/blob/0dabe0d49fa621ed43dc2ba092f56e54d4b117b3/build.zig#L26
There's no need to call this function everytime you invoke git

File: glfw/system_sdk.zig [hunk 2]
```diff
@@ -216,6 +216,7 @@ fn determineSdkRoot(allocator: std.mem.Allocator, org: []const u8, name: []const
         return sdk_root_dir;
     } else |err| return switch (err) {
         error.FileNotFound => {
+            ensureGit(allocator);
```
Comment (iddev5) [inline glfw/system_sdk.zig:219]: And this should be placed here: https://github.com/hexops/mach/blob/f0f91874da865d43eabb751905dd8e0f9026f09e/glfw/system_sdk.zig#L204 as both the code paths use git

Comment (slimsag) [reply glfw/system_sdk.zig:219]: agreed

**Comment by d3m1gd** [comment]
feel free to squash as needed

**Review by iddev5 — APPROVED** [review]

**Review by slimsag — APPROVED** [review]

**Pull request merged**